### PR TITLE
Skip over tokens in the title editor

### LINKED ISSUE #7887: Title Format Editor: Stop allowing the cursor in tokens
> Allowing the user to input text inside the token bubble breaks it.

> Do not show the cursor inside the token bubble, at the beginning nor the end. Two different positions of the cursor have the same effect.

> In fact, I think that we shouldn't show the cursor inside the bubble at all.

from @rodrigoi in…
https://github.com/Automattic/wp-calypso/pull/7313#issuecomment-243953891

### PR DESCRIPTION
**Pull request by dmsnell**
Resolves #7887 

We don't want to allow the cursor to float over tokens in the custom title editor. This PR should intercept editor updates to make sure that we don't allow the cursor to be moved onto a token. This work should prevent accidentally mangling tokens and prevent confusion as to their role.

**Testing**

Navigate to **My Sites** > **Settings** > **SEO** for a site with at least the business plan level and edit the title formats. Attempt to move the cursor into a token. It should skip to the other side of the token.

cc: @roundhill @rodrigoi

### CODE REVIEW

File: client/components/title-format-editor/style.scss [hunk 1]
```diff
@@ -42,9 +42,8 @@
 }
 
 .title-format-editor__token {
-	display: inline-flex;
 	align-items: center;
```
Comment (roundhill) [inline client/components/title-format-editor/style.scss]: `align-items` applies to `flex` only so you can remove it now.

**Comment by roundhill** [comment]
Looks good to go besides the little CSS line you can remove :shipit:

**Comment by dmsnell** [comment]
> the little CSS line you can remove

removed in ac6cc12

**Comment by roundhill** [comment]
> removed in ac6cc12

👍

**Pull request merged**